[CI] DesiredBalanceReconcilerMetricsIT testDesiredBalanceMetrics failing

**Build Scans:**
- [elasticsearch-periodic #4879 / openjdk23_checkpart1_java-matrix](https://gradle-enterprise.elastic.co/s/4dpw5ctkzalmw)
- [elasticsearch-pull-request #41449 / part-1](https://gradle-enterprise.elastic.co/s/7esbctusasn4s)
- [elasticsearch-pull-request #41182 / part-1](https://gradle-enterprise.elastic.co/s/t2ffj5g3kwlc6)

**Reproduction Line:**
```
./gradlew ":server:internalClusterTest" --tests "org.elasticsearch.cluster.routing.allocation.allocator.DesiredBalanceReconcilerMetricsIT.testDesiredBalanceMetrics" -Dtests.seed=B58F2379C103AA4 -Dtests.locale=ks-Arab-IN -Dtests.timezone=Etc/GMT+7 -Druntime.java=23
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.cluster.routing.allocation.allocator.DesiredBalanceReconcilerMetricsIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testDesiredBalanceMetrics),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [main] 3 failures in test testDesiredBalanceMetrics (1.0% fail rate in 294 executions)
- [main] 2 failures in step part-1 (2.0% fail rate in 98 executions)
- [main] 2 failures in pipeline elasticsearch-pull-request (2.0% fail rate in 98 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -240,9 +240,6 @@ tests:
 - class: org.elasticsearch.index.mapper.vectors.MultiDenseVectorScriptDocValuesTests
   method: testFloatGetVectorValueAndGetMagnitude
   issue: https://github.com/elastic/elasticsearch/issues/116863
-- class: org.elasticsearch.cluster.routing.allocation.allocator.DesiredBalanceReconcilerMetricsIT
-  method: testDesiredBalanceMetrics
-  issue: https://github.com/elastic/elasticsearch/issues/116870
 - class: org.elasticsearch.xpack.inference.InferenceRestIT
   method: test {p0=inference/30_semantic_text_inference/Calculates embeddings using the default ELSER 2 endpoint}
   issue: https://github.com/elastic/elasticsearch/issues/116542

--- a/server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconcilerMetricsIT.java
+++ b/server/src/internalClusterTest/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceReconcilerMetricsIT.java
@@ -9,6 +9,7 @@
 
 package org.elasticsearch.cluster.routing.allocation.allocator;
 
+import org.elasticsearch.action.admin.cluster.reroute.ClusterRerouteUtils;
 import org.elasticsearch.cluster.ClusterInfoService;
 import org.elasticsearch.cluster.ClusterInfoServiceUtils;
 import org.elasticsearch.cluster.InternalClusterInfoService;
@@ -68,6 +69,7 @@ public class DesiredBalanceReconcilerMetricsIT extends ESIntegTestCase {
         final var infoService = (InternalClusterInfoService) internalCluster().getCurrentMasterNodeInstance(ClusterInfoService.class);
         ClusterInfoServiceUtils.setUpdateFrequency(infoService, TimeValue.timeValueMillis(200));
         assertNotNull("info should not be null", ClusterInfoServiceUtils.refresh(infoService));
+        ClusterRerouteUtils.reroute(client()); // ensure we leverage the latest cluster info
 
         final var telemetryPlugin = getTelemetryPlugin(internalCluster().getMasterName());
         telemetryPlugin.collect();

